src/shell/components/global-instance/GlobalInstance.js

```diff
@@ -119,17 +120,30 @@ export default connect(state => {
                 setPurge(true);
                 return request(
                   `${CONFIG.CLOUD_FUNCTIONS_DOMAIN}/fastlyPurge?zuid=${props.instance.ZUID}`
-                ).then(res => {
-                  setPurge(false);
-                });
+                )
+                  .then(res => {
+                    if (res.status === 200) {
+                      setPurge(false);
+                    }
+                  })
+                  .catch(err => {
+                    dispatch({
+                      kind: "warn",
+                      message: err.message,
+                      err
+                    });
+                  })
+                  .finally(() => {
+                    setPurge(false);
+                  });
               }}
             >
               {purge ? (
-                <FontAwesomeIcon icon={faSpinner} />
+                <FontAwesomeIcon className={styles.Spinner} icon={faSpinner} />
```

Comment:
We do not need to style this ourselves fontawesome provides a `spin` property which handles this.